tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -553,8 +613,83 @@ void test_sem_take_multiple(void)
 
 	signal_count = k_sem_count_get(&low_prio_sem);
 	zassert_true(signal_count == 1U,
-		     "low priority thread didn't get executed");
+			"Low priority thread don't get the sem");
+}
+
+/**
+ * @brief Test the max value a semaphore can be given and taken
+ * @details
+ * - Reset an initialized semaphore's count to zero.
+ * - Give the semaphore by a thread and verify the semaphore's count whether as
+ *   expected.
+ * - Verify the max count a semaphore can reach.
+ * - Take the semaphore by a thread and verify the semaphore's count whether
+ *   as expected.
+ * - Verify the max times a semaphore can be taken.
+ * @ingroup kernel_semaphore_tests
+ * @see k_sem_count_get(), k_sem_give()
+ */
+void test_k_sem_correct_count_limit(void)
+{
+	uint32_t signal_count;
+	int32_t ret;
+
+	/* reset an initialized semaphore's count to zero */
+	k_sem_reset(&simple_sem);
+	signal_count = k_sem_count_get(&simple_sem);
+	zassert_true(signal_count == 0U, "k_sem_reset failed");
+
+	/* Give the semaphore by a thread and verify the semaphore's
+	 * count whether as expected
+	 */
+	for (int i = 1; i <= SEM_MAX_VAL; i++) {
+		k_sem_give(&simple_sem);
+		signal_count = k_sem_count_get(&simple_sem);
+		zassert_true(signal_count == i,
+				"sem count mismatch expected %d, got %d",
```

Comment:
for readability, may I suggest a symbol between mismatch and expected? 
"sem count mismatch - expected ..."

Reply:
Replaced everywhere.